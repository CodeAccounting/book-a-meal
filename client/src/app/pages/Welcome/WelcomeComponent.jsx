import React, { Fragment } from 'react';
import { HeaderComponent as Header } from '../../shared/Header';
import { FooterComponent as Footer } from '../../shared/Footer';

const message = 'I\'M HUNGRY';

const Welcome = () => (
  <Fragment>
    <Header />
    <div className="main-wrapper home">
      <div className="landing">
        <h1>Delicious Meals At Your Fingertips</h1>
        <a href="./login.html">
          <button className="btn btn-pri">{message}</button>
        </a>
      </div>
    </div>
    <Footer />
  </Fragment>
);

export default Welcome;
