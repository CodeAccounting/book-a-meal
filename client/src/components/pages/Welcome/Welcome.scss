@import '../../../../public/scss/_variables';
.home {
  display: flex;
  flex-direction: column;
  margin-top: 0;
  height: 100%;
  .landing {
    color: $white;
    text-align: center;
    font-family: $heading-font-stack;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 100%;
    h1 {
      margin: 0 0 10%;
      font-size: 2.0em;
      padding: 0 1em;
      margin-bottom: 8.33333%;
      -webkit-animation: pop-in .6s ease-out;
      animation: pop-in .6s ease-out;
    }
    p {
      font-size: 1.5em;
    }
    button {
      border: 2px solid $dark-green;
      background-color: $dark-green;
      color: $white;
      padding: 12px 50px;
      font-size: 1.3em;
      &:hover,
      &.focus,
      &.active {
        transition: background-color 0.5s, border-color 0.5s;
        background-color: #154a19;
        border-color: #154a19;
        color: $white;
      }
    }
  }
  .meals {
    background: linear-gradient(rgba(255, 255, 255, 0.95), rgba(255, 255, 255, 0.95)), url("../../../../public/img/chicken-fajitas.jpg") no-repeat center center;
    background-color: white;
    background-size: cover;
    background-attachment: fixed;
  }
}

@-webkit-keyframes pop-in {
  0% {
    opacity: 0;
    -webkit-transform: translateY(-4rem);
    transform: translateY(-4rem);
  }
  100% {
    opacity: 1;
    -webkit-transform: none;
    transform: none;
  }
}

@keyframes pop-in {
  0% {
    opacity: 0;
    -webkit-transform: translateY(-4rem);
    transform: translateY(-4rem);
  }
  100% {
    opacity: 1;
    -webkit-transform: none;
    transform: none;
  }
}