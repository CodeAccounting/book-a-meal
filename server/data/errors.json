{
  "401": "Unauthorized",
  "403": "Forbidden",
  "404": "Not Found",
  "422": "Validation Error",
  "500": "Internal Server Error",
  "empty": "Empty PUT Requests Are Not Allowed"
}