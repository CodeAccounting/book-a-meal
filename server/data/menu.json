[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "date": "2018-05-06",
     "meals": [
        "81211c24-51c0-46ec-b1e0-18db55880958",
        "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
        "baa0412a-d167-4d2b-b1d8-404cb8f02631",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "date": "2018-04-06",
     "meals": [
        "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
        "3ec802c6-0a32-4d29-b27b-42e0f4b532dd"
     ],
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "date": "2018-05-07",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "date": "2018-04-06",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614",
        "46ced7aa-eed5-4462-b2c0-153f31589bdd"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
     "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
     "date": "2018-04-27",
     "meals": [
        "91b6e41c-0972-4ac5-86da-4ac1f5226e83",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3",
        "8a65538d-f862-420e-bcdc-80743df06578"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  }
]