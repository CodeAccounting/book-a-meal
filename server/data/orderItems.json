[
  {
     "id": 1,
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 2,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
     "id": 2,
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 1,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  }, 
  {
    "id": 3,
    "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
    "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
    "quantity": 2,
    "createdAt": "2018-04-06T14:34:90.000",
    "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
    "id": 4,
    "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  },
  {
    "id": 5,
    "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  }
]