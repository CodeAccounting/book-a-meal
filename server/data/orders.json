[
  {
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "ce228787-f939-40a0-bfd3-6607ca8d2e53",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e544248c-145c-4145-b165-239658857637",
     "created": "2018-05-07T14:34:90.000",
     "updated": "2018-05-07T14:34:90.000",
     "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e5508b87-7975-493d-a900-3d47a69dad03",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  }
]