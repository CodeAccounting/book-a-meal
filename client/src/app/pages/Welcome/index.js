import { connect } from 'react-redux';
import WelcomeComponent from './WelcomeComponent';


const mapStateToProps = state => ({
  authenticating: state.auth.loading,
  isAuthenticated: state.auth.isAuthenticated,
  user: state.auth.user
});

export default connect(mapStateToProps)(WelcomeComponent);
