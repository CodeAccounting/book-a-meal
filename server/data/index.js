import users from './users.json';
import meals from './meals.json';
import menu from './menu.json';
import orders from './orders.json';
import errors from './errors.json';

export default {
  users, meals, menu, orders, errors
};
