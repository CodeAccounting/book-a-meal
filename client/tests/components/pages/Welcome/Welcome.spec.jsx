import React from 'react';
import Welcome from '../../../../src/app/pages/Welcome/WelcomeComponent';
import { caterer } from '../../../setup/data';

const props = {
  user: caterer,
  isAuthenticated: false,
  authenticating: false
};
const { now } = Date;

describe('Welcome', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 0);
  });

  afterAll(() => {
    Date.now = now;
  });

  it('renders correctly when authenticating', () => {
    const shallowWrapper = shallow(<Welcome {...props} authenticating />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('Preloader')).toBeTruthy();
  });

  it('renders correctly when unauthenticated', () => {
    const shallowWrapper = shallow(<Welcome {...props} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('Welcome')).toBeTruthy();
  });

  it('renders correctly when authenticated and user role is caterer', () => {
    const shallowWrapper = shallow(<Welcome {...props} isAuthenticated />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('Dashboard')).toBeTruthy();
  });
});
