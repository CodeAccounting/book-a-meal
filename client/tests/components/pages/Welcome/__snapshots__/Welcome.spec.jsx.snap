// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders correctly when authenticated and user role is caterer 1`] = `
<Connect(Dashboard)
  user={
    Object {
      "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
      "businessName": "Food Circle",
      "businessPhoneNo": "+2348166557788",
      "email": "food@circle.com",
      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
      "role": "caterer",
    }
  }
/>
`;

exports[`Welcome renders correctly when authenticating 1`] = `
<Preloader
  type="user"
/>
`;

exports[`Welcome renders correctly when unauthenticated 1`] = `
<React.Fragment>
  <Header
    dateType={null}
    showTime={true}
    type="home"
  />
  <div
    className="main-wrapper home"
  >
    <div
      className="landing"
    >
      <h1>
        Delicious Meals At Your Fingertips
      </h1>
      <a
        href="/signin"
      >
        <button
          className="btn btn-pri"
        >
          I'M HUNGRY
        </button>
      </a>
    </div>
  </div>
  <Footer />
</React.Fragment>
`;
